src/index.ts

```diff
@@ -381,7 +395,7 @@ export function succeed<A>(a: A): Match<A> {
  * @category matchers
  * @since 0.4.0
  */
-export const end: Match<{}> = new Match(
+export const end: MatchEnd<{}> = new MatchEnd(
```

Comment:
@jleider this implementation will break existing code which uses `end` like that:
```ts
.then(end)
// Throws:
// Argument of type 'MatchEnd<{}>' is not assignable to parameter of type 'Match<{}> & Match<Record<never, never>>'.ts(2345)
```
Even if the package's version is below 1.0.0, the library has been published from quite a lot by now and we have to be as careful as possible